Sources/UtilityBeltNetworking/Models/ResponseValidator.swift

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2021 SpotHero, Inc. All rights reserved.
+
+import Foundation
+
+/// A function that validates an HTTPURLResponse.  Throwing an error indicates a failed validation.
+public typealias ValidationBlock = (HTTPURLResponse) throws -> Void
+
+/// An object that can validate URL responses.
+public struct ResponseValidator {
+    // MARK: Properties
+
+    private let validationCheck: ValidationBlock
+
+    // MARK: Methods
+
+    func validate(response: HTTPURLResponse) throws {
+        try self.validationCheck(response)
+    }
+}
+
+// MARK: - Common Validators
+
+public extension ResponseValidator {
+    /// Creates a validator that will ensure the response has a specific mime type
+    /// - Parameter mimeType: The mime type we want to validate the existance of.
+    /// - Returns: The validator that will do this check.
+    static func ensureMimeType(_ mimeType: MimeType) -> ResponseValidator {
+        return ResponseValidator { response in
+            let responseMimeType = response.mimeType
+            if responseMimeType != mimeType.rawValue {
+                throw UBNetworkError.invalidContentType(responseMimeType ?? "unknown")
+            }
+        }
+    }
+    
+    /// Creates a validator that will ensure the response status code is not an error code.
+    /// - Returns: The validator that will do this check.
+    static let validateStatusCode = ResponseValidator { response in
```

Comment:
I added this validator as well given that we are using the validator pattern in both the combine/non-combine request builders.  I'm not sure I see where this check was being done on the non-combine version?  To be clear, we are adding it in this PR now.

Reply:
If I'm understanding and remembering correctly, we did not validate status codes before, correct?

If not, this may be a breaking change requiring some testing in the app. I think in some ways we rely on a failed status code to then include an error message and some data, so while including this sort of validator is great, I don't know that we should be including this in any defaults.

What are your thoughts?

Reply:
Ya, this is odd.  We did not do any status code checking in the regular request builder, but in the non-combine request builder, we validated status codes on every request.  We probably need some consensus here whether this is something that we mandate or wether we require callers to pass it in. Any thoughts here?  

This also affects how we would handle the `defaultValidators` property - currently in this PR we don't allow overriding them, similar to how the combine request builder just did that validation in the base function (no way to override that as well).  I can see an argument for both...